edunext_openedx_extensions/edunext/management/commands/change_domain.py

```diff
@@ -0,0 +1,83 @@
+"""
+This module contains the command class to change
+the subdomain from prod domains to stage versions.
+"""
+import json
+
+from django.core.management.base import BaseCommand
+
+from edunext_openedx_extensions.ednx_microsites.models import Microsite
+
+
+class Command(BaseCommand):
+
+    """
+    This class contains the methods to change
+    microsite prod domains to a stage versions.
+    """
+    # pylint: disable=unused-argument
+    help = """This function will iterate over all microsites objects
+            to change microsite prod domains to a stage versions."""
+
+    def add_arguments(self, parser):
+        """
+        Add optional domain from the line command
+        """
+        parser.add_argument('suffix_domain', type=str,
+                            nargs='?', default=u"stage.edunext.co")
+
+    def handle(self, *args, **options):
+        """
+        This method will iterate over all microsites
+        objects to change microsite prod domains to a stage versions.
+        """
+
+        # pylint: disable=attribute-defined-outside-init
+        self.suffix_stage_domain = options['suffix_domain']
+
+        # pylint: disable=no-member
+        microsites = Microsite
+
+        for microsite in microsites.objects.all():
+
+            # Don't bother on changing anything if the suffix is correct
+            if microsite.subdomain.endswith(self.suffix_stage_domain):
+                continue
+
+            stage_domain = self.change_subdomain(microsite.subdomain)
+
+            if not stage_domain:
+                continue
+
+            prod_subdomain = microsite.subdomain
+            microsite.subdomain = stage_domain
+
+            configs_string = json.dumps(microsite.values)
+            # Replacing all concidences of the prod domain inside the
+            # configurations by the stage domain
+            modified_configs_string = configs_string.replace(
+                prod_subdomain,
+                stage_domain
+            )
+            microsite.values = json.loads(modified_configs_string)
+            microsite.save()
+
+    def change_subdomain(self, subdomain):
+        """
+        Transforming the domain to format
+        my-microsite-domain-{suffix_stage_domain}
+        """
+
+        try:
+            stage_domain = "{}-{}".format(
+                subdomain.replace('.', '-'),
+                self.suffix_stage_domain
+            )
+        except TypeError as exc:
+            stage_domain = ""
+            message = u"Unable to define stage url for microsite {}".format(
+                subdomain
+            )
+            print message
```

Comment:
can we `LOG.error` the messages instead of plain `print`

Reply:
if we want to give feedback to the user via console, do we use print or not?

Reply:
The correct way is to use a log, then the user can decide to log in the console or a file or whichever available log handler